File: StockFlow.Common/Importer.cs
using Newtonsoft.Json;
using System;
using System.IO;
using System.Text;

namespace StockFlow.Common
{
    public class Importer
    {
        public static void Import<T>(Stream stream, Action<T> importer)
        {
            var reader = new StreamReader(stream, Encoding.UTF8);

            var c = (char)reader.Read();
            if (c == '[')
            {
                ImportList<T>(reader, importer);
            }
            else
            {
                throw new Exception("Should start with [");
            }
        }

        private static void ImportList<T>(StreamReader reader, Action<T> importer)
        {
            while (!reader.EndOfStream)
            {
                var character = reader.Read();

                if (character == '{')
                {
                    ImportListItem<T>(reader, importer);
                }
                else if (character == ']')
                {
                    return;
                }
            }
        }

        private static void ImportListItem<T>(StreamReader reader, Action<T> importer)
        {
            var sb = new StringBuilder("{");
            bool isInQuotes = false;
            int scopes = 1;
            var previousCharacter = (char)0;
            while (!reader.EndOfStream)
            {
                var character = (char)reader.Read();
                sb.Append(character);

                if (previousCharacter != '\\' && character == '\"')
                {
                    isInQuotes = !isInQuotes;
                }
                else if (!isInQuotes && character == '{')
                {
                    scopes++;
                }
                else if (!isInQuotes && character == '}')
                {
                    scopes--;

                    if (scopes == 0)
                    {
                        var json = sb.ToString();
                        var item = JsonConvert.DeserializeObject<T>(json);
                        importer(item);
                        return;
                    }
                }

                previousCharacter = character;
            }
        }

    }
}

File: NAYtrading.Data/process/Importer.py
import json

class Importer:
    def import_stream(self, stream, importer):
        character = stream.read(1)
        if character is None or len(character) == 0:
            return

        if character == '[':
            self.__import_list(stream, importer)
        else:
            raise Exception("Should start with [")

    def __import_list(self, reader, importer):
        while True:
            character = reader.read(1)
            if character is None or len(character) == 0:
                break

            if character == '{':
                self.__import_list_item(reader, importer)
            elif character == ']':
                return

    def __import_list_item(self, reader, importer):
        j = '{'
        is_in_quotes = False
        scopes = 1
        previous_character = '\0'
        while True:
            character = reader.read(1)
            if character is None or len(character) == 0:
                break

            j += character

            if previous_character != '\\' and character == '\"':
                is_in_quotes = not is_in_quotes
            elif (not is_in_quotes) and character == '{':
                scopes += 1
            elif (not is_in_quotes) and character == '}':
                scopes -= 1

                if scopes == 0:
                    item = DictObject(json.loads(j))
                    importer(item)
                    return

            previous_character = character

class DictObject(object):
    def __init__(self, d):
        self.__dict__ = d
        for k in self.__dict__.keys():
            v = self.__dict__[k]
            if isinstance(v, dict):
                self.__dict__[k] = DictObject(v)
            if isinstance(v, list) and len(v) > 0 and isinstance(v[0], dict):
                self.__dict__[k] = [DictObject(x) for x in v]
